### PR TITLE
Implement CloseListing and ModifyListing and some other changes

### PR DESCRIPTION
**Pull request by HellinSun**
This pull request brings the following:

Additions:

- **closeListing** is implemented so that a certain product could be removed. 
- **modifyListing** is implemented so that all the detail information could be modified after the product was created.

- The product page has been updated to show a setting popup window on the top right which allows the user to modify or delete its own products. (Note this is now set to show always for testing, it would be changed to only show on user's own product page later.) An alert dialog would show to ask for user's confirmation again when choosing delete.

- The modifyProduct page now is able to display all the previous information passed from product page and allow the user to modify it based on them.

Changes:

- Add a new public **CameraHelper** class which performs the functionality of taking pictures from camera.
- Changed the product creation/modification format using the **convertDate** method provided by @Lingtao-Cai.
- Made use of Category's **getCategoryById** method created by @artur-varosyan

### CODE REVIEW

File: app/src/main/java/com/example/ar_reshare/BackendController.java [hunk 1]
```diff
@@ -751,5 +733,107 @@ public void run() {
...
+                }
+            });
+        } catch (Exception e) {
+            System.out.println(e);
+        }
+
+    }
+
+    //this is a helper method for addProduct/modifyListing to convert the request body into JSON
+    private static String addOrModifyJsonHelper(Integer productID, String title, String description, String country, String region, String postcode, Integer categoryID, String condition, List<String> media) throws JSONException {
```
Comment (artur-varosyan) [inline app/src/main/java/com/example/ar_reshare/BackendController.java:813]: Love this multifunctional method!

File: app/src/main/java/com/example/ar_reshare/ProductPageActivity.java [hunk 2]
```diff
@@ -147,19 +163,95 @@ public void run() {
...
+                        try {
+                            BackendController.closeListing(product.getId(), new BackendController.BackendCallback() {
+                                @Override
+                                public void onBackendResult(boolean success, String message) {
+                                    if(success){
+                                        Toast.makeText(getApplicationContext(),
+                                                "Product deleted successfully!",
+                                                Toast.LENGTH_LONG).show();
+                                        // go back to the main page when finished
+                                        startActivity(new Intent(ProductPageActivity.this, ARActivity.class));
```
Comment (artur-varosyan) [inline app/src/main/java/com/example/ar_reshare/ProductPageActivity.java:237]: Are you sure going back to main activity after deleting the item is the best choice? Should we maybe go to profile page? What does everyone think?

**Review by artur-varosyan — APPROVED** [review]
Great functionality and very well documented!

**Review by Manoutofpower — APPROVED** [review]

File: app/src/main/java/com/example/ar_reshare/ProductPageActivity.java (hunk 2 above)
Comment (arfi2212) [reply app/src/main/java/com/example/ar_reshare/ProductPageActivity.java:237]: I feel like we should go to the profile page as well!

**Review by arfi2212 — APPROVED** [review]
Lovely work :)

**Review by Lingtao-Cai — APPROVED** [review]
👍

File: app/src/main/java/com/example/ar_reshare/ProductPageActivity.java (hunk 2 above)
Comment (HellinSun) [reply app/src/main/java/com/example/ar_reshare/ProductPageActivity.java:237]: yeah that makes more sense, thanks guys! I'll change it once profile is ready

**Pull request merged**